### PR TITLE
(next-urql) - Pass resetUrqlClient method to App/Page

### PR DESCRIPTION
**Pull request by JoviDeCroock**
## Summary

Resolves: https://github.com/FormidableLabs/urql/issues/832

We can rename this to `_dangerous_reset_store` since this resets all the exchange states, ....

## Set of changes

- Add function to invalidate current client instance

### CODE REVIEW

**Comment by zvictor** [comment]
I am getting `Error: Invalid hook call` once I switch to this branch.

image

It looks like React doesn't recognise the code within `withUrqlClient` as a react context.

**Comment by JoviDeCroock** [comment]
@zvictor that's because you have two react instances you can use the build I used here to circumvent that. https://codesandbox.io/s/next-urql-pokedex-2ophq?file=/components/pokemon_list.js

File: packages/next-urql/README.md [hunk 1]
```diff
@@ -121,6 +121,14 @@ withUrqlClient((_ssrExchange, ctx) => ({
 
 In client-side SPAs using `urql`, you typically configure the Client yourself and pass it as the `value` prop to `urql`'s context `Provider`. `withUrqlClient` handles setting all of this up for you under the hood. By default, you'll be opted into server-side `Suspense` and have the necessary `exchanges` set up for you, including the [`ssrExchange`](https://formidable.com/open-source/urql/docs/api/#ssrexchange-exchange-factory). If you need to customize your exchanges beyond the defaults `next-urql` provides, use the second argument to `withUrqlClient`, `mergeExchanges`.
 
+### Resetting the client instance
+
+In rare scenario's you possibly will have to reset the client instance (reset all cache, ...), this is an uncommon scenario
+and we consider it "unsafe" so evaluate this carefully for yourself.
+
+When this does seem like the appropriate solution any component wrapped with `withUrqlClient` will receive the `resetClient`
```
Comment (kitten) [inline packages/next-urql/README.md:129]:
```suggestion
When this does seem like the appropriate solution any component wrapped with `withUrqlClient` will receive the `resetUrqlClient`
```

File: docs/advanced/server-side-rendering.md [hunk 2]
```diff
@@ -240,3 +240,11 @@ export default withUrqlClient(ssrExchange => ({
 
 Unless the component that is being wrapped already has a `getInitialProps` method, `next-urql` won't add its own SSR logic, which automatically fetches queries during
 server-side rendering. This can be explicitly enabled by passing the `{ ssr: true }` option as a second argument to `withUrqlClient`.
+
+### Resetting the client instance
+
+In rare scenario's you possibly will have to reset the client instance (reset all cache, ...), this is an uncommon scenario
+and we consider it "unsafe" so evaluate this carefully for yourself.
+
+When this does seem like the appropriate solution any component wrapped with `withUrqlClient` will receive the `resetClient`
```
Comment (kitten) [inline docs/advanced/server-side-rendering.md:249]:
```suggestion
When this does seem like the appropriate solution any component wrapped with `withUrqlClient` will receive the `resetUrqlClient`
```

**Pull request merged**